src/azure-cli-core/azure/cli/core/__init__.py

```diff
@@ -559,6 +546,93 @@ def load_arguments(self, command=None):
                 self.extra_argument_registry.update(loader.extra_argument_registry)
                 loader._update_command_definitions()  # pylint: disable=protected-access
 
+    def _load_modules(self, args, command_modules):
+        """Load command modules using ThreadPoolExecutor with timeout protection."""
+        from azure.cli.core.commands import BLOCKED_MODS
+
+        results = []
+        with ThreadPoolExecutor(max_workers=MAX_WORKER_THREAD_COUNT) as executor:
+            future_to_module = {executor.submit(self._load_single_module, mod, args): mod
+                                for mod in command_modules if mod not in BLOCKED_MODS}
+
+            for future in concurrent.futures.as_completed(future_to_module):
+                try:
+                    result = future.result(timeout=MODULE_LOAD_TIMEOUT_SECONDS)
+                    results.append(result)
+                except concurrent.futures.TimeoutError:
+                    mod = future_to_module[future]
+                    logger.warning("Module '%s' load timeout after %s seconds", mod, MODULE_LOAD_TIMEOUT_SECONDS)
+                    results.append(ModuleLoadResult(mod, {}, {}, 0,
+                                                    Exception(f"Module '{mod}' load timeout")))
+                except (ImportError, AttributeError, TypeError, ValueError) as ex:
+                    mod = future_to_module[future]
+                    logger.warning("Module '%s' load failed: %s", mod, ex)
+                    results.append(ModuleLoadResult(mod, {}, {}, 0, ex))
+                except Exception as ex:  # pylint: disable=broad-exception-caught
+                    mod = future_to_module[future]
+                    logger.warning("Module '%s' load failed with unexpected exception: %s", mod, ex)
+                    results.append(ModuleLoadResult(mod, {}, {}, 0, ex))
+
+        return results
+
+    def _load_single_module(self, mod, args):
+        from azure.cli.core.breaking_change import import_module_breaking_changes
+        from azure.cli.core.commands import _load_module_command_loader
+        import traceback
+        try:
+            start_time = timeit.default_timer()
+            module_command_table, module_group_table = _load_module_command_loader(self, args, mod)
```

Comment:
Is there a risk of race conditions when loading `command_loader` with multiple threads? How can we ensure that all loaders are properly registered and that each command is correctly mapped to its corresponding loader?
https://github.com/Azure/azure-cli/blob/11568f4e46d45b2c20bc74b269993b57ddb10e4a/src/azure-cli-core/azure/cli/core/commands/__init__.py#L1137
https://github.com/Azure/azure-cli/blob/11568f4e46d45b2c20bc74b269993b57ddb10e4a/src/azure-cli-core/azure/cli/core/commands/__init__.py#L1146